### PR TITLE
Fix URI handling in mleap; test mleap on Windows

### LINKED ISSUE #6958: Issue While Serializing Model When Logging With Mlflow Mleap
I am facing an issue when logging my model through the mlflow.mleap.log_model function. Below are my following settings:

I created an experiment with the local file path set as the artifact_location argument.

mlflow.create_experiment\
    (
        name="decision_tree",
        artifact_location="file:/D:/Mlflow/artifacts",
        tags={"version":"v1","priority":"p1"},
    )
  
I used the spark model sample code written on the [mlflow mleap](https://www.mlflow.org/docs/latest/python_api/mlflow.mleap.html) page. 

image

However, when I call the mlflow.mleap method, it raises the following issue:

image

I have tried to change the artifact location to different local path files and even HDFS path, but the same issue occurs. Kindly look into this and let me know if it is an issue from my side.

### PR DESCRIPTION
**Pull request by dbczumar**
## Related Issues/PRs

Fixes #6958 

## What changes are proposed in this pull request?

Fix URI handling in mleap; test mleap on Windows

## How is this patch tested?

Run mleap tests on Windows

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

## Release Notes

Fix a pathing issue with MLeap model persistence on Windows. 

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by BenWilson2 — APPROVED** [review]
LGTM!

File: mlflow/mleap.py [hunk 1]
```diff
@@ -243,7 +244,8 @@ def add_to_model(mlflow_model, path, spark_model, sample_input):
     os.makedirs(mleap_path_full)
 
     dataset = spark_model.transform(sample_input)
-    model_path = "file:{mp}".format(mp=mleap_datapath_full)
+    model_path = path_to_local_file_uri(mleap_datapath_full)
+    model_path = model_path.replace("file:///", "file:/")
```
Comment (dbczumar) [inline mlflow/mleap.py:248]:
```suggestion
    model_path = model_path.replace("file:///", "file:")
```

File: mlflow/mleap.py [hunk 2]
```diff
@@ -243,7 +244,10 @@ def add_to_model(mlflow_model, path, spark_model, sample_input):
     os.makedirs(mleap_path_full)
 
     dataset = spark_model.transform(sample_input)
-    model_path = "file:{mp}".format(mp=mleap_datapath_full)
+    if os.name != "nt":
+        model_path = path_to_local_file_uri(mleap_datapath_full)
+    else:
+        model_path = mleap_datapath_full
```
Comment (BenWilson2) [inline mlflow/mleap.py:250]: Does the mleap Bundler require the prefix 'file:/'? https://github.com/combust/mleap/blob/a2dbcee6860802da2fda7f4b14e1dd35129214aa/bundle-ml/src/main/scala/ml/combust/bundle/BundleFile.scala#L22-L37

Comment (dbczumar) [reply mlflow/mleap.py:250]: I've tried `file:`, `file:/`, and `file:///`, none of which work. Will dig in further to see if there's some alternative structure for Windows.

Comment (dbczumar) [reply mlflow/mleap.py:250]: I've tried `file:`, `file:/`, and `file:///`, none of which work. Will dig in further to see if there's some alternative structure for Windows.

**Pull request merged**